### DISCUSSION TITLE
Phrase triggering of workflows in pull requests

### DISCUSSION THREAD

**Discussion by mwalenia** [community/community#25543] [Actions]
<p>Hi,</p>
<p>I’m trying to duplicate behaviour of Jenkins ghprb plugin, which allows you to start a Jenkins job by posting a trigger phrase in a pull request comment.</p>
<p>I think that the closest event to use in Actions would be </p>
<pre><code>pull_request_review_comment
</code></pre>
<p>but I didn’t manage to use it so far.</p>
<p>Does anyone know how to do that?</p>
<p>Cheers!</p>

**Comment by tyankatsu0105** [comment]
<p>I have the same issue.</p>
<p>I’m creating actions these days.</p>
<p><a href="https://github.com/tyankatsu0105/message-to-slack-from-github/pull/1" rel="nofollow noopener">https://github.com/tyankatsu0105/message-to-slack-from-github/pull/1</a></p>
<p>But I can’t find a solution to the trigger for PR comment. </p>
<p>I found this page. <a href="https://developer.github.com/v3/guides/working-with-comments/#pull-request-comments" rel="nofollow noopener">https://developer.github.com/v3/guides/working-with-comments/#pull-request-comments</a></p>
<p>According to this page, PR comments use issues API.</p>
<p>I tried adding <code>issue_comment</code> on <code>on</code>.</p>
<p>But not working…</p>

**Comment by tyankatsu0105** [comment]
<p>OK.</p>
<p>I understand.</p>
<p>PR message’s eventName is <code>issue_comment</code>.</p>
<p>So you can check <code>Mention Slack</code> jobs’s log in here.</p>
<p><a href="https://github.com/tyankatsu0105/message-to-slack-from-github/commit/9a72de4917a22696b0ece5cd6ca9c11cfc2d21c8/checks?check_suite_id=334701535" rel="nofollow noopener">https://github.com/tyankatsu0105/message-to-slack-from-github/commit/9a72de4917a22696b0ece5cd6ca9c11cfc2d21c8/checks?check_suite_id=334701535</a></p>

**Comment by dnorth98** [comment]
<p>Having the same issue.  I’ve created a small workflow that just prints the comment but it never fires when I comment on the pull request containing this workflow.  Even tried covering all the bases for the different events but no joy.</p>
<p>Do we somehow have to register the workflow to fire on PR comments some other way?</p>
<pre><code>name: terraform-staging-apply

on:
  issue_comment:
    types: [created, edited]
  pull_request_review_comment:
    types: [created, edited]
  pull_request_review:
    types: [submitted, edited, dismissed]
  issues:
    types: [opened, closed]

jobs:
  terraform-apply:
    name: Terraform Apply
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@1.0.0
        with:
          submodules: 'true'
          token: ${{ secrets.PAT }} # Needed for private submodules

      - uses: actions/github-script@0.2.0
        with:
          github-token: ${{github.token}}
          script: |
            const body = context.payload.pull_request.body.toLowerCase()
            console.log("Detected comment: " + body)
</code></pre>

**Comment by dnorth98** [comment, marked as answer]
<p>Found the issue in another thread:</p>
<p><a href="https://github.community/t/on-issue-comment-events-are-not-triggering-workflows/16784">https://github.community/t5/GitHub-Actions/on-issue-comment-events-are-not-triggering-workflows/m-p/37177</a></p>
<p>In short, the workflow triggering on issue comments MUST be in master</p>
